core/src/initializers/telemetry.ts

```diff
@@ -10,7 +11,13 @@ export class TelemetryInitializer extends CLSInitializer {
   }
 
   async initializeWithinTransaction() {}
-  async startWithinTransaction() {}
+
+  async startWithinTransaction() {
+    if (getGrouparooRunMode() === "cli:config") {
+      await ConfigUser.loadOrStoreCustomerId();
```

Comment:
```suggestion
      await ConfigUser.loadAndStoreCustomerId();
```

Reply:
you might want to give this initializer a `startPriority` of a low number so this happens before other steps